### PR TITLE
HW25

### PR DESCRIPTION
**Pull request by Artsapollo**

### CODE REVIEW

File: src/main/java/lesson22UsingHibernate/OrderModelDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    public Set<OrderModel> getAllOrders() {
+        Set<OrderModel> orderModel = new HashSet<>();
+        try {
+            sessionObj = buildSessionFactory().openSession();
+            sessionObj.beginTransaction();
+            orderModel = new HashSet<>(sessionObj.createQuery("FROM OrderModel").list());
+        } catch (Exception sqlException) {
+            if (null != sessionObj.getTransaction()) {
+                LOG.info("\n...Transaction failed...");
+                sessionObj.getTransaction().rollback();
```
Comment (Cherkashyn) [inline src/main/java/lesson22UsingHibernate/OrderModelDaoImpl.java]: We don't insert/update/delete anything here. Rollback unnecesary here.

File: src/main/java/lesson22UsingHibernate/OrderModelDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+    @Override
+    public boolean deleteOrderModel(BigDecimal id) {
+        boolean result = false;
+
+        try {
+            sessionObj = buildSessionFactory().openSession();
+            sessionObj.beginTransaction();
+
+            sessionObj.delete(sessionObj.get(OrderModel.class, id));
```
Comment (Cherkashyn) [inline src/main/java/lesson22UsingHibernate/OrderModelDaoImpl.java]: Why u use "get" instead "find"?

File: src/main/java/lesson22UsingHibernate/OrderModelDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    }
+
+    @Override
+    public boolean updateOrderModel(OrderModel orderModel) {
+        boolean result = false;
+        try {
+            sessionObj = buildSessionFactory().openSession();
+            sessionObj.beginTransaction();
+
+            OrderModel om = sessionObj.get(OrderModel.class, orderModel.getOrderNum());
```
Comment (Cherkashyn) [inline src/main/java/lesson22UsingHibernate/OrderModelDaoImpl.java]: I think .getOrderNum don't need here.

File: src/test/java/lesson22UsingHibernate/OrderModelDaoImplTest.java [hunk 4]
```diff
@@ -42,4 +31,15 @@ public void testUpdateOrder() {
...
+    @Test
+    public void getAllOrders() {
+        assertNotNull(orderModelDao.getAllOrders());
+    }
+
+    @Test
+    public void findOrderById() {
+        OrderModel tutu = orderModelDao.findOrderModelById(new BigDecimal(112961));
+        assertNotNull(tutu);
+    }
```
Comment (Cherkashyn) [inline src/test/java/lesson22UsingHibernate/OrderModelDaoImplTest.java]: And what about insert/update/delete tests? Where are they? :)

File: src/main/java/lesson22UsingHibernate/OrderModelDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+                sessionObj.close();
+            }
+        }
+        return orderModel;
+    }
+
+    @Override
+    public OrderModel findOrderModelById(BigDecimal id) {
+        OrderModel orderModel = null;
+        try {
```
Comment (SergeyPasko) [inline src/main/java/lesson22UsingHibernate/OrderModelDaoImpl.java]: Make sence doing same logic to getAllOrders

File: src/main/java/lesson22UsingHibernate/OrderModelDaoImpl.java (hunk 3 above)
Comment (SergeyPasko) [reply src/main/java/lesson22UsingHibernate/OrderModelDaoImpl.java]: It depends your orderModel under JPA control or not

File: src/main/java/lesson22UsingHibernate/OrderModelDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+    public boolean updateOrderModel(OrderModel orderModel) {
+        boolean result = false;
+        try {
+            sessionObj = buildSessionFactory().openSession();
+            sessionObj.beginTransaction();
+
+            OrderModel om = sessionObj.get(OrderModel.class, orderModel.getOrderNum());
+            om.setAmount(om.getAmount());
+            om.setQty(om.getQty());
+
```
Comment (SergeyPasko) [inline src/main/java/lesson22UsingHibernate/OrderModelDaoImpl.java]: no any operation in result

File: src/main/java/lesson22UsingHibernate/OrderModelDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+            sessionObj = buildSessionFactory().openSession();
+            sessionObj.beginTransaction();
+            orderModel = sessionObj.load(OrderModel.class, id);
+        } catch (Exception sqlException) {
+            if (null != sessionObj.getTransaction()) {
+                LOG.info("\n...Transaction failed...");
+                sessionObj.getTransaction().rollback();
+            }
+            sqlException.printStackTrace();
+        }
```
Comment (SergeyPasko) [inline src/main/java/lesson22UsingHibernate/OrderModelDaoImpl.java]: you didn't close session

File: src/main/java/lesson22UsingHibernate/OrderModelDaoImpl.java [hunk 8]
```diff
@@ -96,9 +96,7 @@ public boolean insertOrderModel(OrderModel orderModel) {
             }
             sqlException.printStackTrace();
         } finally {
-            if (sessionObj != null) {
```
Comment (SergeyPasko) [inline src/main/java/lesson22UsingHibernate/OrderModelDaoImpl.java]: if you have exception on cretion session your sessionObj  will be null. And you in sessionObj.close() will be have NullPointerException

**Pull request merged**